kubee2etests/helpers_and_globals.py

```diff
@@ -110,3 +112,52 @@ def load_kubernetes():
             logging.error("Not able to use in-cluster config: %s", err)
             sys.exit(1)
     return incluster
+
+def add_error(self, err):
+    error_list = [error[0] for error in self.errors]
+    try:
+        idx = error_list.index(err)
+        self.errors[idx] = (err, self.errors[idx][1] + 1)
+    except ValueError:
+        self.errors.append((err, 1))
+
+def add_errors(self, errors):
+    """
+    Method which adds errors to the object's error list - this allows us to batch send them
+    when a status update is needed. It also does deduping of errors which have come up twice
+    since the last status was sent to the frontend.
+
+    Args:
+        errors: (list) list of tuples to add to the error list - first entry being the error, second being the number of occurences
+
+    Returns: None, as a side effect will update `self.errors`
+
+    """
+    error_list = [error[0] for error in self.errors]
+    for err in errors:
+        try:
+            idx = error_list.index(err[0])
+            self.errors[idx] = (err, self.errors[idx][1] + err[1])
+        except ValueError:
+            self.errors.append(err)
+
+def send_update(self, name):
+    """
```

Comment:
I'd change this to just have it take `results` as an argument, shouldn't be passing self around if it's outside of a class